test/views/WhatDoYouWantToDoViewSpec.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import models.Availability
+import play.twirl.api.HtmlFormat
+import views.behaviours.ViewBehaviours
+import views.html.WhatDoYouWantToDoView
+
+class WhatDoYouWantToDoViewSpec extends ViewBehaviours {
```

Comment:
Might actually make sense to split this out a bit, so have something like:

```
def applyView(availability: Availability) = injector.instanceOf[WhatDoYouWantToDoView].apply(availability, availability)(fakeRequest, messages)

"when departures and arrivals empty must" - {
    behave like ...
}

"when departures and arrivals unavailable must" - {
    val doc = parseView(applyView(Unavailable))
    behave like ...
}

"when departures and arrivals non-empty must" - {
    val doc = parseView(applyView(NonEmpty))
    behave like ...
}
```

Reply:
Could then also have something like

```
private val availability: Availability = arbitrary[Availability].sample.value

override def view: HtmlFormat.Appendable =
    applyView(availability, availability)
```

so we know that the behave like `pageWithTitle()`, `behave like pageWithBackLink()` etc. tests aren't dependent on the availability type

Reply:
Looks good, I'll do that, but I'm keeping Arrivals and Departures separate to stop the potential of cut and past issues meaning that the departure logic uses the arrivals availability

Reply:
Good idea